### PR TITLE
warp: refactor warp handler stats

### PR DESCRIPTION
**Pull request by aaronbuchwald**
This PR refactors warp handler stats to get rid of an unnecessary interface and mock stats implementation.

### CODE REVIEW

File: warp/handlers/stats.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+type handlerStats struct {
+	// SignatureRequestHandler metrics
+	signatureRequest        metrics.Counter
+	signatureHit            metrics.Counter
+	signatureMiss           metrics.Counter
+	signatureProcessingTime metrics.Gauge
+}
+
+func newStats() *handlerStats {
```
Comment (joshua-kim) [inline warp/handlers/stats.go:20]: would personally not use a pointer/pointer receiver here but its a super nit

**Review by joshua-kim — APPROVED** [review]

File: warp/handlers/signature_request_test.go [hunk 2]
```diff
@@ -64,27 +61,28 @@ func TestSignatureHandler(t *testing.T) {
...
 		},
 	}
 
 	for name, test := range tests {
-		// Reset stats before each test
-		mockHandlerStats.Reset()
-
 		t.Run(name, func(t *testing.T) {
+			handler := NewSignatureRequestHandler(backend, message.Codec)
+			handler.stats.Clear()
```
Comment (ceyonur) [inline warp/handlers/signature_request_test.go:76]: isn't this already clean?

File: warp/handlers/stats.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	return &handlerStats{
+		signatureRequest:        metrics.GetOrRegisterCounter("signature_request_count", nil),
+		signatureHit:            metrics.GetOrRegisterCounter("signature_request_hit", nil),
+		signatureMiss:           metrics.GetOrRegisterCounter("signature_request_miss", nil),
+		signatureProcessingTime: metrics.GetOrRegisterGauge("signature_request_duration", nil),
```
Comment (ceyonur) [inline warp/handlers/stats.go:25]: nit: can we rename either the variable or metric name?

Comment (aaronbuchwald) [reply warp/handlers/stats.go:25]: Ya I'll rename the variable

File: warp/handlers/signature_request_test.go (hunk 2 above)
Comment (aaronbuchwald) [reply warp/handlers/signature_request_test.go:76]: No, these are globally registered, so we need to clear after each run.

File: warp/handlers/stats.go (hunk 3 above)
Comment (aaronbuchwald) [reply warp/handlers/stats.go]: Renamed

**Review by ceyonur — APPROVED** [review]

**Review by abi87 — APPROVED** [review]

**Review by anusha-ctrl — APPROVED** [review]

**Pull request merged**